### DISCUSSION TITLE
Latest Android, can't empty cache for Bluetooth.

### DISCUSSION THREAD

**Discussion by Headlands7** [NightscoutFoundation/xDrip#2587] [General]
On the latest Android version, and suddenly a few days ago my S22 will not see the G6 transmitter, so xDrip+ is permanently on "G6 State Unknown". The lovely Google software and S22 can't find it/don't see it -- it's not an xDrip+ issue, but an Android issue. 

Normally the thing that would fix this would be to clear the cache of Bluetooth on the phone, but Google has now disabled this feature. So I'm without xDrip+ and need to use my Omnipod 5 to see my glucose levels (the G6 works fine with that with Bluetooth). Again: You can't empty the cache for Bluetooth anymore on the most current Android OS._

This is NOT an xDrip+ issue, but I want to hear if any fellow xDrip+ users here know how to fix this, whether it's on S22 or other Android (I've read tons of posts saying this is the same on other Android phones with the very latest Android software). If you don't lose the Bluetooth connection to the G6 then you're fine, but if you do then you can't empty the cache anymore to fix it...in that case the phone doesn't see the G6 anymore and I'm dearly hoping there's some workaround.

**Comment by Navid200** [comment]
If you look at your phone, can you see any details about the Android patch?  Any versions?  Can you post a screenshot?
If this is something that is going affect many, the sooner we get all the info, the better we can identify and hopefully find a remedy for the problem.

**Reply by Headlands7** [reply]
Sure. The first one shows the greyed out clear cache and clear data, the second two are software version (I don't know what to make of the info)
![Screenshot_20221229_091146_Settings](https://user-images.githubusercontent.com/57432113/209987017-5fc0ec06-9d3a-4bb2-aa33-33add873b485.jpg)
![Screenshot_20221229_091428_Settings](https://user-images.githubusercontent.com/57432113/209987051-2cb27cf1-b5d6-418c-ba73-0528686f1b92.jpg)
![Screenshot_20221229_091432_Settings](https://user-images.githubusercontent.com/57432113/209987052-91ddca03-9e72-4589-82d6-bc5f1d3008eb.jpg)

**Comment by fdiazdiaz** [comment]
Hi, same issue here with a Samsung S20 FE after Dec update. Bluetooth cache is greyed out so I can't empty it. I emptied the cache of Bluetooth Agent and Bluetooth MIDI service, it looked it worked but after 30 min got stuck again. Attaching the screenshot of patched app and other Bluetooth services 
![inbound1436396684672204314](https://user-images.githubusercontent.com/80062626/209983267-5be2aa6e-b1c8-487a-8150-441210004484.jpg)
![inbound5439004742674656291](https://user-images.githubusercontent.com/80062626/209983287-be8e662b-8112-4942-a268-1a561235b0a6.jpg)

**Comment by Navid200** [comment]
Thanks for the info.
I understand that you cannot empty the Bluetooth cache.  

But, is there a problem using xDrip now?  As far as I know, there is no requirement for the xDrip user to clear the cache.
I have been using xDrip for more than 4 years now and I have never cleared Bluetooth cache on my phone.

**Reply by Headlands7** [reply]
Yes, that's why I wrote that it's not an xDrip+ issue, but that I was hoping someone here might have some insight on how to remedy it since we're all on Android and the only thing it's affecting is my G6.

**Reply by Navid200** [reply]
I have been using xDrip and G6 for the past 4 years.  I never need to clear the Bluetooth cache.  Why do you?

**Reply by Headlands7** [reply]
When an Android isn't communicating well with the G6 and you get the "G6 state unknown" on xDrip+ because of it, one great and easy way to fix it (which I found about here on this very forum!) is to clear the Bluetooth cache. Now that option is not available anymore with the latest Android update. I've been using xDrip+ for about as long as you.

**Reply by Navid200** [reply]
As you have stated already, that option seems to be gone.  Does it really make any difference how long you have been using it?
Specially knowing that others don't even need it?

The date on that patch is December 1.  It is very likely that many other xDrip users have received that patch by now.

These are the settings I use:  https://navid200.github.io/xDrip/docs/G6-Recommended-Settings
I don't get frequent disconnects.  If I do get one, I restart the collector and it reconnects.

**Reply by Headlands7** [reply]
There is some misinterpretation going on here, but all good, that's the the nature of forums. :-) It's not an xDrip+ issue as I've said, but thanks for the help.
